RUMM-2590 Update distributed tracing on ios (#15503)
DataDog/documentation: content/en/real_user_monitoring/connect_rum_and_traces/_index.md

2.  Set the `firstPartyHosts` initialization parameter with the list of internal, first-party origins called by your iOS application.
    ```swift
    Datadog.initialize(
        appContext: .init(),
        configuration: Datadog.Configuration
            .builderUsing(rumApplicationID: "<rum_app_id>", clientToken: "<client_token>", environment: "<env_name>")
            .set(firstPartyHosts: ["example.com", "api.yourdomain.com"])
            .build()
    )
    ```

3. Initialize the global `Tracer`:
    ```swift
    Global.sharedTracer = Tracer.initialize(
        configuration: Tracer.Configuration(...)
    )
    ```

> 4.  Initialize URLSession as stated in [Setup][1]:

Reviewer: ```suggestion
4. Initialize URLSession as stated in [Setup][1]:
```

> **Note**: By default, all subdomains of listed hosts are traced. For instance, if you add `example.com`, you also enable tracing for `api.example.com` and `foo.example.com`.

Reviewer: ```suggestion
By default, all subdomains of listed hosts are traced. For instance, if you add `example.com`, you also enable tracing for `api.example.com` and `foo.example.com`.
```

> **Note 2**: Trace ID injection works when providing a `URLRequest` to the `URLSession`, distributed tracing will not work when using a `URL` object.

Reviewer: ```suggestion
Trace ID injection works when you are providing a `URLRequest` to the `URLSession`. Distributed tracing does not work when you are using a `URL` object.
```

> 5.  _(Optional)_ Set the `tracingSamplingRate` initialization parameter to keep a defined percentage of the backend traces. If not set, 100% of the traces coming from application requests are sent to Datadog. To keep 20% of backend traces:

Reviewer: ```suggestion
5.  Optionally, set the `tracingSamplingRate` initialization parameter to keep a defined percentage of the backend traces. If not set, 100% of the traces coming from application requests are sent to Datadog. 

     To keep 20% of backend traces:
```
